src/wp-includes/cache.php

```diff
@@ -102,6 +102,45 @@ function wp_cache_flush() {
 
 	return $wp_object_cache->flush();
 }
+/**
+ * Removes all cache items in a group.
+ *
+ * @since 6.0.0
+ *
+ * @see WP_Object_Cache::flush_group()
+ * @global WP_Object_Cache $wp_object_cache Object cache global instance.
+ *
+ * @param string|array $group name(s) of group to remove from cache.
+ *
+ * @return bool True on success, false on failure group not found.
+ */
+function wp_cache_flush_group( $group ) {
+	global $wp_object_cache;
+
+	// if group is an array loop and call each key in the array
+	if ( is_array( $group ) ) {
+		foreach ( $group as $group_key ) {
+
+			wp_cache_flush_group( $group_key );
+		}
+
+		return true;
+	}
+
+	if ( method_exists( $wp_object_cache, 'flush_group' ) ) {
+		// these are linked cache groups, so we have to flush them both if one is called
+		if ( 'users' === $group ) {
+			$wp_object_cache->flush_group( 'user_meta' );
+		}
+		if ( 'user_meta' === $group ) {
+			$wp_object_cache->flush_group( 'users' );
+		}
+
+		return $wp_object_cache->flush_group( $group );
+	}
```

Comment:
I am not sure I understand this. Why also check these link groups?

If we are going to do this, then, all meta group would need to be cleared. So post, comment, site and term. Not to mention other linked groups, like network and site (network) options. If we do this, we would need a way to register the links together, so that developers could link custom groups together.

Reply:
I found a comment about this linked group and felt it was safer for the core to handle linked groups rather than rely on coders knowing (I didn't) about the links
So yes I believe we should handle all the linked groups I just don't know them all

Reply:
and I should change the code to use an array to hold the mappings

Reply:
I have refactored this to allow to adding additional mappings 
@spacedmonkey can you help with the pairs we need to flush together